code.R

```diff
@@ -0,0 +1,238 @@
+# As you can see, the plot is skewed to the right.
+tidy_galaxies %>%
+  filter(variable == "m_b") %>% 
+  ggplot(aes(x = value)) + 
+  geom_histogram(
+    aes(y = ..density..),
+    colour = "#e9ecef",
+    alpha = 0.6,
+    binwidth = 1,
+    na.rm = TRUE) +
+  stat_function(fun = dnorm,
+                args = list(
+                  mean = stats_magnitudes$mean, 
+                  sd = stats_magnitudes$sd
+                ),
+                colour = "red",
+                geom = "point"
+  ) +
+  ylab("Relative frequency") +
+  xlab("Absolute magnitude of the galaxy") +
+  scale_y_continuous(labels = scales::percent_format(accuracy = 1)) +
+  theme_minimal() +
+  ggtitle("Relative frequencies of galaxy magnitudes", 
+          "Data taken from 'UPDATED NEARBY GALAXY CATALOG', 
+          The Astronomical Journal, 145:101 (22pp), 2013 April")
+
+
+# We can also calculate skewness and kurtosis.
+# The distribution of magnitudes is skewed to the
+# right and has fatter tails.
+tidy_galaxies %>% 
+  filter(variable == "m_b") %$% 
+  c(
+    skewness = moments::skewness(value),
+    kurtosis = moments::kurtosis(value)
+  )
+
+# An explanation for this could be that galaxies with bigger
+# magnitudes (the light emitted by the galaxy) are easier to
+# observe, as the telescopes might not be sensitive enough
+# for galaxies with little light emission.
+
+
+
+###### TASK 4 ######
+
+# Load data ----
+galaxy_speeds <-
+  readLines("https://www.sao.ru/lv/lvgdb/article/UCNG_Table4.txt") %>% 
+  as_tibble()
+
+# Deal with separating line ----
+sep_line_speeds <- galaxy_speeds %>% 
+  findSeparatingLine()
+
+galaxy_speeds <-
+  galaxy_speeds %>% 
+  slice(-sep_line_speeds)
+
+# Separate the columns of the table ----
+
+# Get column names
+column_names_speeds <-
+  galaxy_speeds %>% 
+  getColumnNames()
+
+# Separate into columns
+galaxy_speeds_table <- 
+  galaxy_speeds %>% 
+  separateColumns(column_names_speeds)
+
+# Convert columns containing numbers into numeric
+galaxy_speeds_table <-
+  galaxy_speeds_table %>% 
+  mutate(across(c(cz, error), as.numeric))
+
+# Again, we want to save the character columns into a
+# separate tibble, as tidying the table would
+# result in a character value column
+
+galaxy_speeds_table %>% 
+  select(name, bibcode, refs) -> speed_references
+
+tidy_speeds <- galaxy_speeds_table %>% 
+  select(name, cz, error) %>% 
+  melt(id.vars = "name") %>% 
+  as_tibble()
+
+# Join velocity to galaxy_table ----
+
+# Extract name and cz column
+velocity_table <-
+  tidy_speeds %>% 
+  filter(variable == "cz")
+
+# Join
+tidy_galaxies <- 
+  tidy_galaxies %>% 
+  left_join(velocity_table, by = join_by(name == name))
+
+
+# Plot velocity against distance ----
+tidy_galaxies %>% 
+  filter(variable.x == "D") %>% 
+  ggplot(aes(x = value.x, y = value.y)) +
+  geom_point() +
+  geom_smooth(method = lm)
+
+# Yes, there seems to be a roughly proportional relationship.
+
+# Estimate Hubble's constant as H = v/D ----
+hubble_constant <-
```

Comment:
Hubbles contant can be estimated running a linear regression with Velocity as dependent variable, Distance as independent variable and removing the Intercept.